internal/elements.go

```diff
@@ -417,6 +418,16 @@ type CurrentUserPrincipal struct {
 	Unauthenticated *struct{} `xml:"unauthenticated,omitempty"`
 }
 
+// https://datatracker.ietf.org/doc/html/rfc3744#section-5.4
+type CurrentUserPrivilegeSet struct {
+	XMLName    xml.Name    `xml:"DAV: current-user-principal"`
+	Privileges []Privilege `xml:"privilege"`
+}
+type Privilege struct {
+	Read  bool `xml:"read"`
+	Write bool `xml:"write"`
+}
```

Comment:
In my case, I got it to work with Thunderbird locally with the following structure (didn't tried yours TBH):
```suggestion
	Privileges []Privilege `xml:"privilege>dynamic"`
}
type Privilege struct {
	XMLName xml.Name
}
```

And then return `read`, `write` like this:
```go
return &internal.CurrentUserPrivilegeSet{Privileges: []internal.Privilege{{XMLName: xml.Name{"", "read"}}, {XMLName: xml.Name{"", "write"}}}}, nil
```

(ideally the read/write privileges information should come from the backend)

Reply:
Seems like the second part of the suggestion hasn't been applied. (But I prefer the struct approach.)